elb_automation/meaoelb/elb_ctx.py

```diff
@@ -0,0 +1,173 @@
+            LoadBalancerName=elb_config.name,
+            Listeners=list(map(lambda l: l.to_aws(), elb_config.listeners)),
+            SecurityGroups=elb_config.security_groups,
+            Subnets=elb_config.subnets,
+            Tags=elb_config.tags)
+        print("Done")
+
+    def _modify_elb_atts(self, service_config):
+        elb_config = service_config.elb_config
+        if elb_config.elb_atts:
+            print("\t➤ Updating ELB attributes...", end='', flush=True)
+            response = self.elb_client.modify_load_balancer_attributes(
+                LoadBalancerName=elb_config.name,
+                LoadBalancerAttributes=elb_config.elb_atts.to_aws())
+            print("Done")
+        else:
+            print("➤ No ELB attributes defined")
+
+    def _configure_elb_health_checks(self, service_config):
+        elb_config = service_config.elb_config
+        print("\t➤ Creating health checks...", end='', flush=True)
+        hc = elb_config.health_check
+        hc_target = "{}:{}{}".format(
+            hc.target_proto,
+            hc.target_port,
+            hc.target_path)
+        response = self.elb_client.configure_health_check(
+            LoadBalancerName=elb_config.name,
+            HealthCheck={
+                'Target': hc_target,
+                'Interval': hc.interval,
+                'Timeout': hc.timeout,
+                'UnhealthyThreshold': hc.unhealthy_threshold,
+                'HealthyThreshold': hc.healthy_threshold
+            })
+        print("Done")
+
+    def prompt_for_apply(self):
+        if not self.confirmed_apply:
+            result = input("Enter 'make it so' to continue: ")
+            if result != 'make it so':
+                raise Exception("User cancelled operation")
+            else:
+                self.confirmed_apply = True
+
+    def create_elb(self, service_config):
+        """
+        Create an ELB, modify it's attributes, and create health checks.
+        """
+        if self.get_cluster_name() != service_config.target_cluster:
+            print(
+                "➤ Currently connected to {} K8s cluster, which is not the target: {}".format(
+                    self.get_cluster_name(),
+                    service_config.target_cluster))
+            return
+
+        if not self.dry_run_mode and not self.confirmed_apply:
+            self.prompt_for_apply()
+
+        print("➤ Processing {}".format(service_config.elb_config.name))
+        if self.elb_exists(service_config.elb_config.name):
+            print(
+                "\t➤ {} has already been provisioned".format(
+                    service_config.elb_config.name))
+            return
+
+        if self.dry_run_mode:
+            print(
+                "\t➤ ELB {} would have been created".format(
+                    service_config.elb_config.name))
+            return
+
+        self._create_elb(service_config)
+        self._modify_elb_atts(service_config)
```

Comment:
Why do we create then immediately modify the elb, instead of creating the ELB with the desired attributes? Is this working around some limitation of the AWS api?

Reply:
great question - it's because the [create_load_balancer](https://boto3.readthedocs.io/en/latest/reference/services/elb.html#ElasticLoadBalancing.Client.create_load_balancer) call doesn't take any attributes :-(

Reply:
Answering my own question: I see from https://docs.aws.amazon.com/elasticloadbalancing/2012-06-01/APIReference/API_CreateLoadBalancer.html that there doesn't appear to be a way to pass in https://docs.aws.amazon.com/elasticloadbalancing/2012-06-01/APIReference/API_LoadBalancerAttributes.html so we have to use https://docs.aws.amazon.com/elasticloadbalancing/2012-06-01/APIReference/API_ModifyLoadBalancerAttributes.html instead.